### PR TITLE
[16][FIX] rma: Refactor onchange fields to computed fields (partial)

### PR DESCRIPTION
**Pull request by florian-dacosta**
This is still kind of WIP. I think the tests on RMA pass because the Form way to create rma is not really used in the tests.

There is still refactore todo, most of the onchange should be gone...
I am not sure if I will have time to complete these kind of refactores, but feels free to superseed this PR of course.

### CODE REVIEW

**Comment by LoisRForgeFlow** [comment]
@florian-dacosta I just faced some issues that would be solved with this refactor, I may supersede this PR. If you have done more work in this direction, please let me know to not do it twice :)

**Comment by florian-dacosta** [comment]
Hello, nothing more on my side, for me we can merge it already and make additional PR or if you prefer superseed, it is ok for me too of course!

**Comment by LoisRForgeFlow** [comment]
For now I have handled the problem I had with this https://github.com/ForgeFlow/stock-rma/pull/355. My original problem was in v14 and I couldn't take advantage of the precompute yet. The change has been [forward ported as is to v16](https://github.com/ForgeFlow/stock-rma/pull/358), but it would be greate to refactor.

**Comment by florian-dacosta** [comment]
@LoisRForgeFlow I don't think I'll have time to make a refactore of all onchange fields any time soon.
But I think we could remove the WIP tag and merge this PR, and keep accepting PR about these kind of refactore, to make it little by little.
Would it be acceptable ?

File: rma/models/rma_order_line.py [hunk 1]
```diff
@@ -704,7 +721,12 @@ def _onchange_lot_id(self):
         product = self.lot_id.product_id
         if product:
             self.product_id = product
-            self.uom_id = product.uom_id
+
+    @api.depends("uom_id")
+    def _compute_uom_id(self):
+        for rec in self:
+            if rec.product_id:
+                rec.uom_id = rec.product_id.uom_id
```
Comment (LoisRForgeFlow) [inline rma/models/rma_order_line.py]: uom_id compute depends on "uom_id"? this doesn't look correct

**Review by LoisRForgeFlow — COMMENTED** [review]
Hi @florian-dacosta 

I have nothing against merging this, however could first review my comment bellow?

File: rma/models/rma_order_line.py (hunk 1 above)
Comment (florian-dacosta) [reply rma/models/rma_order_line.py]: It is not right indeed, fixed, thanks

Comment (LoisRForgeFlow) [reply rma/models/rma_order_line.py]: Thanks! I will fix pre-commit in a different pr if needed.

**Review by LoisRForgeFlow — APPROVED** [review]

**Pull request merged**